src/Utilities/functions.jl

```diff
@@ -1816,6 +1816,15 @@ end
 
 ### ScalarNonlinearFunction
 
+# Needed for LessToGreaterBridge
+function promote_operation(
+    ::Union{typeof(+),typeof(-)},
+    ::Type{<:Real},
+    ::Type{MOI.ScalarNonlinearFunction},
+)
+    return MOI.ScalarNonlinearFunction
+end
```

Comment:
@blegat: so one issue. A lot of the bridges are very permissive in the functions that accept. And then they use `promote_operation` to figure out what the resultant function is. That makes it hard for new functions, because we need to add a bunch of these `promote_operation` methods. An alternative would be to make the bridge explicitly declare what functions they support, and external packages would need to opt-into them. Not something we necessarily need to change now, but another MOI 2.0 list.

Reply:
Having to opt-in for each bridge would loose a lot of the bridges advantages. Right now, the fact that we will be able to no nonlinear conic with the bridges just by implementing a few `MOI.Utilities` is quite appealing.
I can understand that when you create a new function you would like to be able to test it without having to do all these things.
One thing you could do is to use `MOI.Bridges.remove_bridge` with the bridges that are annoying you.
We could have a way to say that an operation is not supported as well so that you could do that and then not have to implement `MOI.Utilities.operate`, etc...